### DISCUSSION TITLE
revisionHistoryLimit with Rook Helm Chart

### DISCUSSION THREAD

**Discussion by reefland** [rook/rook#12693] [Q&A] [answered]
I'm trying to determine where the correct location would be to place a `revisionHistoryLimit: 3` in the Rook Operator and Cluster helm charts.

- Use ArgoCD / Git to manage deployment history
- Rook is deployed via Helm Chart
- Will never use revisionHistory to rollback deployments when using Helm and Git

The default is 10, so every pod / replicaset -- every plugin, ceph-tools, crash collector, mds, rgw, mgr, operator, every OSD will maintain 10 copies.  Interacting with `rook-ceph` namespace either via `kubectl` and especially a GUI like ArgoCD is a cumbersome - it has to gather and render all this information.

Would like to reduce the history amount to making this more manageable.  Having two previous copies is plenty for troubleshooting / check for changes between releases.

I reviewed the Helm templates, didn't see anything obvious being used.  Before I open an issue for a feature request, figured I'd ask here.

**Comment by travisn** [comment]
10 does seem like a high default, perhaps it even makes sense to reduce the default to 3 when implementing this in Rook. 

I'm thinking we would need the following changes:
- Add a setting to the CephCluster CR for `revisionHistoryLimit`. Rook would apply this setting to all deployments created (mon, mgr, osd, rgw, mds, etc). However, since this affects the CSI driver components, perhaps this should instead be an overall setting in the `rook-ceph-operator-config` configmap. It does seem better suited for the configmap as an overall consistency setting for the operator. Thoughts?
- The helm chart will expose the setting (from the previous bullet) in values.yaml
- Wherever there is a `DeploymentSpec` in the code, the `RevisionHistoryLimit` would be set according to that cephcluster setting.
- Evaluate the setting in the csi deployment templates under `pkg/operator/ceph/csi/template`

**Reply by reefland** [reply]
Should I open a feature request with above information?

**Reply by travisn** [reply, marked as answer]
Yes sounds good to open an issue for the feature request, thanks!
